tldw_Server_API/app/core/DB_Management/media_db/legacy_reads.py

```
@@ -0,0 +1,180 @@
+        with db_instance.transaction() as conn:
+            rows = db_instance._fetchall_with_connection(conn, query, (media_id,))
+    except (DatabaseError, sqlite3.Error) as exc:
+        logger.exception(
+            f"Error getting transcripts media {media_id} '{db_instance.db_path_str}'"
+        )
+        raise DatabaseError(f"Failed get transcripts {media_id}") from exc  # noqa: TRY003
+    else:
+        return rows
+
+
+def get_latest_transcription(
```

Ruff (0.15.6)

[warning] Too many return statements (8 > 6)

(PLR0911)